src/main.c

```diff
@@ -0,0 +1,222 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/sys/byteorder.h>
+
+// Added this so devices would be remembered
+#include <zephyr/settings/settings.h>
+
+static void start_scan(void);
+
+static struct bt_conn *conn_connecting = NULL;
+static uint8_t volatile conn_count;
+static bool volatile is_connecting = false;
+
+static bool find_device_name(struct bt_data *data, void *user_data)
+{
+    char* device_name = user_data;
+
+    // a return value of "false" will stop data parsing by bt_data_parse
+
+    if (data->type == BT_DATA_NAME_COMPLETE) {
+        memcpy(device_name, data->data, data->data_len);
+        device_name[data->data_len] = '\0';
+        return false;
+    }
+    else {
+        return true;
+    }
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	int err;
+    char device_name[31];
+    memset(device_name, 0, sizeof(device_name));
+
+	if (conn_connecting) {
+		return;
+	}
+
+	/* We're only interested in connectable events */
+	if (type != BT_GAP_ADV_TYPE_ADV_IND &&
+	    type != BT_GAP_ADV_TYPE_ADV_DIRECT_IND) {
+		return;
+	}
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+	printk("Device found: %s (RSSI %d)\n", addr_str, rssi);
+
+    // bt_data_parse accepts a buffer of the ad data, 
+    // a callback function, and a place to store the parsed data
+
+    bt_data_parse(ad, find_device_name, (void *)device_name);
```

Comment:
@jakejunk150 We should handle the `bt_conn_le_create` inside `bt_data_parse`. Can you update it? 
Besides that, can you filter with the device name contains `DXC` and create the connect when you find it only?

Reply:
@trungkiena6 Is it okay if I handle the `bt_conn_le_create` inside my callback function (`find_device_name`), or should I edit the source code for `bt_data_parse`?

Reply:
@jakejunk150 I refer to handle the `bt_conn_le_create` inside your callback function. It would be better than modifying the Zephyr source. Also, that is how we manage user API in Stel :).